### PR TITLE
Added JSON import functionality in scripts

### PR DESCRIPTION
**Pull request by AshanthaLahiru**
### Add ability to import a JSON as a constant

Feature:
```
in this context
   JARVIS_JSON is from "./json/sample.json"
end
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -100,6 +101,25 @@ class Jarvis {
     return null;
   }
 
+  /**
+   * Validate the constant format and store the constant in memory
+   */
+  _setConstant(key, value) {
```
Comment (hliyan) [inline index.js]: clarify whether this is for all constants or just .json

File: index.js [hunk 2]
```diff
@@ -100,6 +101,25 @@ class Jarvis {
...
+    // get the top most value of the stack which is currently active script
+    const currentScriptPath = this.importStack[this.importStack.length - 1];
+
+    if (this.constants[key]) {
```
Comment (hliyan) [inline index.js]: move this above line 109

File: index.js [hunk 3]
```diff
@@ -100,6 +101,25 @@ class Jarvis {
...
+      return `'${key}' constant already exists!`;
+    }
+    else if (key === key.toUpperCase()) {
+      let constant = { key, value };
+      this.activeContext[currentScriptPath].push(constant);
+      return;
+    } else {
```
Comment (hliyan) [inline index.js]: make this a guard clause and move to the top

File: src/utils.js [hunk 4]
```diff
@@ -1,11 +1,19 @@
 // jarvis, just another rudimentary verbal interface shell
-// converts 'hello "John Doe"' to ['hello', 'John, Doe']
 const fs = require("fs");
 const path = require("path");
+
+// checks whether the given argument is in the format of a stringified JSON
+const isJson = (argument) => {
+  return /<({.+})>/.test(argument);
```
Comment (hliyan) [inline src/utils.js]: how is this json?

File: src/utils.js [hunk 5]
```diff
@@ -1,11 +1,19 @@
...
+}
+
+// converts 'hello "John Doe" <{"name": "JARVIS"}>' to ['hello', 'John Doe',  <{"name": "JARVIS"}>]
 const tokenize = line => {
-  const tokens = line.match(/"([^"]+)"|\S+/g);
+  const tokens = line.match(/"?<({.+})>"?|"([^"]+)"|\S+/g);
```
Comment (hliyan) [inline src/utils.js]: what's the purpose of the change?

File: src/utils.js (hunk 4 above)
Comment (AshanthaLahiru) [reply src/utils.js]: Stringified  JSON is wrapped by the `<` `>` to make it more robust while tokenizing. That's the reason for the above format validation to check whether the input format is in the JSON format.

File: src/utils.js (hunk 5 above)
Comment (AshanthaLahiru) [reply src/utils.js]: As mentioned above, parsed string is wrapped by  "`<>`" to make it more robust while tokenizing.

**Comment by AshanthaLahiru** [comment]
Moved to #26
